akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.impl.engine.http2
+
+import akka.stream.Attributes
+import akka.stream.BidiShape
+import akka.stream.Inlet
+import akka.stream.Outlet
+import akka.stream.stage.GraphStage
+import akka.stream.stage.GraphStageLogic
+
+/**
+ * This stage contains all control logic for handling frames and (de)muxing data to/from substreams.
+ *
+ * (This is not a final documentation, more like a brain-dump of how it could work.)
+ *
+ * The BidiStage consumes and produces FrameEvents from the network. It will output one Http2SubStream
+ * for incoming frames per substream and likewise accepts a single Http2SubStream per substream with
+ * outgoing frames.
```

Comment:
Exactly, or http messages without entities by putting a Source.empty inside the Http2SubStream.